OrbitLinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -164,21 +44,10 @@ void UprobesUnwindingVisitor::visit(UprobesWithStackPerfEvent* event) {
                                         event->GetFunction()->VirtualAddress(),
                                         event->GetTimestamp());
 
-  const std::vector<unwindstack::FrameData>& callstack = unwinder_.Unwind(
-      event->GetRegisters(), event->GetStackData(), event->GetStackSize());
-  const std::vector<unwindstack::FrameData>& full_callstack =
-      callstack_manager_.ProcessUprobesCallstack(event->GetTid(), callstack);
-
-  // TODO: Callstacks at the beginning and/or end of a dynamically-instrumented
-  //  function could alter the statistics of time-based callstack sampling.
-  //  Consider not/conditionally adding these callstacks to the trace.
-  if (!full_callstack.empty()) {
-    Callstack returned_callstack{
-        event->GetTid(),
-        CallstackFramesFromLibunwindstackFrames(full_callstack),
-        event->GetTimestamp()};
-    listener_->OnCallstack(returned_callstack);
-  }
+  // Careful: UprobesWithStackPerfEvent* event ends up being moved from
+  // LateUnwindCallstack's constructor.
+  callstack_manager_.ProcessUprobesCallstack(event->GetTid(),
+                                             std::move(*event));
```

Comment:
Should the visitor interface be changed as well? To reflect the fact that whatever is passed to visit() method is lost afterwards.

Reply:
I don't like the idea too much because it would make for a weird interface for a visitor. After all each visitor method already takes a pointer, which tells that the pointee could be modified.